python-lib/sharepoint_lists.py

```diff
@@ -97,8 +97,8 @@ def __init__(self, config, parent, dataset_schema, dataset_partitioning, partiti
         self.web_name = self.parent.sharepoint_list_title
 
         if write_mode == SharePointConstants.WRITE_MODE_CREATE:
-            logger.info('flush:delete_list "{}"'.format(self.parent.sharepoint_list_title))
-            self.parent.client.delete_list(self.parent.sharepoint_list_title)
+            logger.info('flush:recycle_list "{}"'.format(self.parent.sharepoint_list_title))
+            self.parent.client.recycle_list(self.parent.sharepoint_list_title)
```

Comment:
The delete_list() function from sharepoint_client file is not used anymore, could be removed

Reply:
👍 Done in [0012d1e](https://github.com/dataiku/dss-plugin-sharepoint-online/pull/38/commits/0012d1edc0289da9ac62f319f40f76b438034717)